feat: P4ADEV-2795 created debt position and installment entities and repositories

dan-ntt opened the pull request:
#### Description
The goal of this pull request is to define a structure for the entities of debt position and installment registries.

#### List of Changes
- defined a base registry entity
- defined an event type enum so that all known events are properly managed and easily detectable
- defined entity and repository for debt position registry
- defined entity and repository for instalment registry

#### How Has This Been Tested?
- Pre-Deploy Test
  - [ ] Unit
  - [ ] Integration (Narrow)
- Post-Deploy Test
  - [ ] Isolated Microservice
  - [ ] Broader Integration
  - [ ] Acceptance
  - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [x] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] CHORE - Minor Change (fix or feature that don't impact the functionality e.g. Documentation or lint configuration)

#### Checklist:

Commit:
feat(P4ADEV-2795): created base registry entity and event type enum

Commit:
feat(P4ADEV-2795): created main entities and repositories
- Debt Position Registry entity
- Debt Position Registry repository
- Installment Registry entity
- Installment Registry repository

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/enums/EventType.java:
why have you created this enum?
you should re-use it.gov.pagopa.pu.workflowhub.dto.generated.PaymentEventType

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/model/DebtPositionRegistry.java:
```diff
@@ -0,0 +1,24 @@
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class DebtPositionRegistry extends BaseRegistry implements Serializable {
+  private Long debtPositionId;
+  private Long organizationId;
```
these 2 fields whould belong also to `InstallmentRegistry`

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/model/BaseRegistry.java:
```diff
@@ -0,0 +1,28 @@
+@SuperBuilder
+@EqualsAndHashCode
+@ToString
+public abstract class BaseRegistry {
```
Considering that will exist also `sil_registry` and `pagopa_registry`, which will not share this structure, I would not use such generic name...
Moreover, I cannot see a real advantage on having a common base class to share common fields... I would not create this class

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/model/BaseRegistry.java:
```diff
@@ -0,0 +1,28 @@
+  @Id
+  private String eventId;
+  private EventType eventType;
+  private LocalDateTime eventDateTime;
```
As you could see from PaymentEventDTO class, this is an OffsetDateTime, while the enum should not be re-created, but you should use the same used in PaymentEventDTO
```suggestion
  private OffsetDateTime eventDateTime;
```

Commit:
fix(P4ADEV-2795): resolved PR comments
- removed EventType and made use of PaymentEventType
- removed base entity
- added missing fields to installment
- changed LocalDateTime with OffsetDateTime for eventDateTime prop

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/repository/DebtPositionRegistryRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource(collectionResourceRel = "debtPositionRegistries", path = "debt-position-registry")
```
the path should be in the plural form 
```suggestion
@RepositoryRestResource(collectionResourceRel = "debtPositionRegistries", path = "debt-position-registries")
```

dan-ntt replied:
I've picked "registry" because in this case it's a "registry" of installment events and not multiple registries of installment events - made the change

antonioT90 replied:
we use the plural form for the part in order to be more Restfull
eg:
GET /.../...registries to fetch all

Commit:
fix(P4ADEV-2795): updated openapi specifications

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/repository/InstallmentRegistryRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource(collectionResourceRel = "installmentRegistries", path = "installment-registry")
```
the path should be in the plural form
```suggestion
@RepositoryRestResource(collectionResourceRel = "installmentRegistries", path = "installment-registries")
```

dan-ntt replied:
I've picked "registry" because in this case it's a "registry" of installment events and not multiple registries of installment events - made the change

Commit:
fix(P4ADEV-2795): changed names to plural

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/model/InstallmentRegistry.java:
```diff
@@ -0,0 +1,34 @@
+import java.io.Serializable;
+import java.time.OffsetDateTime;
+
+@Document(collection = "debt_position_registry")
```
```suggestion
@Document(collection = "installment_registry")
```

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/repository/DebtPositionRegistryRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource(collectionResourceRel = "debtPositionRegistries", path = "debt-position-registries")
```
is this really needed?
```suggestion
@RepositoryRestResource(path = "debt-position-registries")
```

antonioT90 commented on src/main/java/it/gov/pagopa/pu/registry/repository/InstallmentRegistryRepository.java:
```diff
@@ -0,0 +1,9 @@
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+
+@RepositoryRestResource(collectionResourceRel = "installmentRegistries", path = "installment-registries")
```
is this really needed?
```suggestion
@RepositoryRestResource(path = "installment-registries")
```

Commit:
fix(P4ADEV-2795): removed some information on repositories, corrected collection name missmatch

Commit:
Merge branch 'develop' into P4ADEV-2795-implementation-base-entities

Commit:
fix(P4ADEV-2795): updated openapi
